java/test/jmri/util/JUnitUtil.java

```diff
@@ -238,7 +238,10 @@ public static void setUp() {
         if(initPrefsDir==null){
            initPrefsDir = System.getProperty("jmri.prefsdir");
         }
-        System.setProperty("jmri.prefsdir",initPrefsDir);
+        
+        if (initPrefsDir != null) {
```

Comment:
@pabender this change appears to re-introduce the problems you were trying to solve with 2feecc401e65766e90a8a6b168a06a1e806f8eb6 in #7105. Does it?

Reply:
Yes, I think it does reintroduce that problem.  It certainly allows tests to be run without the property being defined, which can’t be the case if the tests are run via ant or Maven.

Reply:
If  initPrefsDir is null, then there's an exception.  So given the tests did run correctly, one can assume that it wasn't null.  Not sure why my configuration sees it as null, but not setting the property doesn't affect testing on my system.

Reply:
So what is the proper fix?

Reply:
Looking at how we use the setProperty in our code, it seems to me that the line:

initPrefsDir = System.getProperty("jmri.prefsdir");

isn't quire right, I believe initPrefsDir should be a path name to a directory.

Reply:
You are running the tests natively in Eclipse right?  If that is the case, you need to have Eclipse set the system properly.

I don’t know how to make that happen.  I just know that both ant and Maven set the property.

Reply:
InitPrefsDir IS a directory, but it is set via an system property.

https://github.com/JMRI/JMRI/blob/4cd5deb29b449bb4da47d473ea1322991d993cd6/build.xml#L762

Reply:
Now understood.  Thanks Paul.